presto-spark-base/src/main/java/com/facebook/presto/spark/execution/common/WorkerConfigProperties.java

```diff
@@ -0,0 +1,92 @@
+ */
+public class WorkerConfigProperties
+        extends Properties
+{
+    public static final String CONCURRENT_LIFESPANS_PER_TASK = "concurrent-lifespans-per-task";
+    public static final String ENABLE_SERIALIZED_PAGE_CHECKSUM = "enable-serialized-page-checksum";
+    public static final String ENABLE_VELOX_EXPRESSION_LOGGING = "enable_velox_expression_logging";
+    public static final String ENABLE_VELOX_TASK_LOGGING = "enable_velox_task_logging";
+    public static final String HTTP_SERVER_HTTP_PORT = "http-server.http.port";
+    public static final String HTTP_EXEC_THREADS = "http_exec_threads";
+    public static final String NUM_IO_THREADS = "num-io-threads";
+    public static final String PRESTO_VERSION = "presto.version";
+    public static final String SHUTDOWN_ONSET_SEC = "shutdown-onset-sec";
+    public static final String SYSTEM_MEMORY_GB = "system-memory-gb";
+    public static final String TASK_MAX_DRIVERS_PER_TASK = "task.max-drivers-per-task";
+    public static final String DISCOVERY_URI = "discovery.uri";
+    public static final String STORAGE_ONCALL_NAME = "storage_oncall_name";
+    public static final String STORAGE_USER_NAME = "storage_user_name";
+    public static final String STORAGE_SERVICE_NAME = "storage_service_name";
+
+    private static final String DEFAULT_CONCURRENT_LIFESPANS_PER_TASK = "5";
+    private static final String DEFAULT_ENABLE_SERIALIZED_PAGE_CHECKSUM = "true";
+    private static final String DEFAULT_ENABLE_VELOX_EXPRESSION_LOGGING = "false";
+    private static final String DEFAULT_ENABLE_VELOX_TASK_LOGGING = "true";
+    private static final String DEFAULT_HTTP_SERVER_HTTP_PORT = "7777";
+    private static final String DEFAULT_HTTP_EXEC_THREADS = "128";
+    private static final String DEFAULT_NUM_IO_THREADS = "30";
+    private static final String DEFAULT_PRESTO_VERSION = "0.277.fb-1.cpp-3";
+    private static final String DEFAULT_SHUTDOWN_ONSET_SEC = "10";
+    private static final String DEFAULT_SYSTEM_MEMORY_GB = "10";
+    private static final String DEFAULT_TASK_MAX_DRIVERS_PER_TASK = "15";
+    private static final String DEFAULT_DISCOVERY_URI = "http://127.0.0.1";
+    private static final String DEFAULT_STORAGE_ONCALL_NAME = "presto_cpp";
+    private static final String DEFAULT_STORAGE_USER_NAME = "presto";
+    private static final String DEFAULT_STORAGE_SERVICE_NAME = "presto.native";
+
+    private WorkerConfigProperties(PropertiesPopulator populator, ImmutableMap<String, String> propertiesMap)
+    {
+        super(populator, propertiesMap);
+    }
+
+    public static class WorkerConfigPropertiesBuilder
+            extends Properties.PropertiesBuilder<WorkerConfigPropertiesBuilder>
+    {
+        public WorkerConfigPropertiesBuilder initializeDefault()
+        {
+            propertiesMap.put(CONCURRENT_LIFESPANS_PER_TASK, DEFAULT_CONCURRENT_LIFESPANS_PER_TASK);
```

Comment:
Since the `propertiesMap` is meat to be immutable, should we just use `setProperty(key, value)` here? Same in other propertiesBulders

Reply:
maps in builders have to be mutable. But yeah we can do setProperty calls instead of direct calls to map.